refactor(oxfmt): simplify `wrap_format_embeded`

Dunqing opened the pull request:
`call_async` does the same thing as `call_with_return_value` + `channel`, so use `call_async` to simplify it and reduce some unnecessary overhead

### Benchmark

I benchmarked this PR locally, and the results indicate that it is 9% faster than before in the `Outline` repo.

EDIT: Maybe just a noise since `Warning: Statistical outliers were detected. Consider re-running this benchmark on a quiet system without any interference from other programs.` was displayed.

```shell

Setup complete! Run 'pnpm run bench' to start benchmarking.
=========================================
JavaScript/TypeScript Formatter Benchmark
=========================================

Formatters: Prettier, Biome, Oxfmt

Starting benchmark with:
- 3 warmup runs
- 10 benchmark runs
- Git reset before each run


=========================================
Benchmarking parser.ts (single large file)
=========================================
Benchmark 1: biome
  Time (mean ± σ):      69.5 ms ±   0.9 ms    [User: 55.4 ms, System: 9.7 ms]
  Range (min … max):    68.3 ms …  71.2 ms    10 runs
 
Benchmark 2: oxfmt-new
  Time (mean ± σ):      30.7 ms ±   0.3 ms    [User: 23.7 ms, System: 11.0 ms]
  Range (min … max):    30.3 ms …  31.3 ms    10 runs
 
Benchmark 3: oxfmt-old
  Time (mean ± σ):      30.4 ms ±   0.3 ms    [User: 23.6 ms, System: 10.6 ms]
  Range (min … max):    30.1 ms …  30.9 ms    10 runs
 
Summary
  oxfmt-old ran
    1.01 ± 0.01 times faster than oxfmt-new
    2.29 ± 0.04 times faster than biome

=========================================
Benchmarking Outline repository
=========================================
Benchmark 1: biome
  Time (mean ± σ):     154.6 ms ±   6.0 ms    [User: 825.2 ms, System: 555.8 ms]
  Range (min … max):   147.5 ms … 164.0 ms    10 runs
 
Benchmark 2: oxfmt-new
  Time (mean ± σ):      65.8 ms ±   2.3 ms    [User: 196.0 ms, System: 265.3 ms]
  Range (min … max):    60.7 ms …  68.8 ms    10 runs
 
Benchmark 3: oxfmt-old
  Time (mean ± σ):      72.0 ms ±  17.7 ms    [User: 195.0 ms, System: 277.3 ms]
  Range (min … max):    62.6 ms … 121.1 ms    10 runs
 
  Warning: Statistical outliers were detected. Consider re-running this benchmark on a quiet system without any interferences from other programs.
 
Summary
  oxfmt-new ran
    1.09 ± 0.27 times faster than oxfmt-old
    2.35 ± 0.12 times faster than biome
```

Dunqing commented:
* **#15798** <a href="https://app.graphite.com/github/pr/oxc-project/oxc/15798?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.com/github/pr/oxc-project/oxc/15798?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.com/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=15798).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

leaysgur commented:
Just FYI, you may or may not know:

- The `outline` repository contains 26 files that include ``css`...` ``
- But there are only 4 of them do not contain substitution
  - = cases where the current `oxc_formatter` calls Prettier

So while the benchmark might not be very reliable, I still think this approach is better. 😅

leaysgur commented:
### Merge activity

* **Nov 18, 6:28 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Nov 18, 6:28 AM UTC**: `leaysgur` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

Dunqing committed:
refactor(oxfmt): simplify `wrap_format_embeded` (#15798)

`call_async` does the same thing as `call_with_return_value` + `channel`, so use `call_async` to simplify it and reduce some unnecessary overhead

### Benchmark

I benchmarked this PR locally, and the results indicate that it is 9% faster than before in the `Outline` repo.

EDIT: Maybe just a noise since `Warning: Statistical outliers were detected. Consider re-running this benchmark on a quiet system without any interference from other programs.` was displayed.

```shell

Setup complete! Run 'pnpm run bench' to start benchmarking.
=========================================
JavaScript/TypeScript Formatter Benchmark
=========================================

Formatters: Prettier, Biome, Oxfmt

Starting benchmark with:
- 3 warmup runs
- 10 benchmark runs
- Git reset before each run

=========================================
Benchmarking parser.ts (single large file)
=========================================
Benchmark 1: biome
  Time (mean ± σ):      69.5 ms ±   0.9 ms    [User: 55.4 ms, System: 9.7 ms]
  Range (min … max):    68.3 ms …  71.2 ms    10 runs

Benchmark 2: oxfmt-new
  Time (mean ± σ):      30.7 ms ±   0.3 ms    [User: 23.7 ms, System: 11.0 ms]
  Range (min … max):    30.3 ms …  31.3 ms    10 runs

Benchmark 3: oxfmt-old
  Time (mean ± σ):      30.4 ms ±   0.3 ms    [User: 23.6 ms, System: 10.6 ms]
  Range (min … max):    30.1 ms …  30.9 ms    10 runs

Summary
  oxfmt-old ran
    1.01 ± 0.01 times faster than oxfmt-new
    2.29 ± 0.04 times faster than biome

=========================================
Benchmarking Outline repository
=========================================
Benchmark 1: biome
  Time (mean ± σ):     154.6 ms ±   6.0 ms    [User: 825.2 ms, System: 555.8 ms]
  Range (min … max):   147.5 ms … 164.0 ms    10 runs

Benchmark 2: oxfmt-new
  Time (mean ± σ):      65.8 ms ±   2.3 ms    [User: 196.0 ms, System: 265.3 ms]
  Range (min … max):    60.7 ms …  68.8 ms    10 runs

Benchmark 3: oxfmt-old
  Time (mean ± σ):      72.0 ms ±  17.7 ms    [User: 195.0 ms, System: 277.3 ms]
  Range (min … max):    62.6 ms … 121.1 ms    10 runs

  Warning: Statistical outliers were detected. Consider re-running this benchmark on a quiet system without any interferences from other programs.

Summary
  oxfmt-new ran
    1.09 ± 0.27 times faster than oxfmt-old
    2.35 ± 0.12 times faster than biome
```
